packages/codec/lib/basic/decode/index.ts

```diff
@@ -306,22 +310,23 @@ export function* decodeBasic(
             value: yield* decodeExternalFunction(address, selector, info)
           };
         case "internal":
-          if (strict) {
-            //internal functions don't go in the ABI!
-            //this should never happen, but just to be sure...
-            throw new StopDecodingError({
-              kind: "InternalFunctionInABIError" as const
-            });
-          }
+          //note: we no longer error here if strict is true, because we
+          //want to be able to use strict mode to decode immutables, and
+          //these can be immutables.  yes, this is a bit of an abuse of
+          //strict mode, which was meant for ABI decoding, but oh well.
           if (!checkPadding(bytes, dataType, paddingMode)) {
+            const error = {
+              kind: "FunctionInternalPaddingError" as const,
+              paddingType: getPaddingType(dataType, paddingMode),
+              raw: Conversion.toHexString(bytes)
+            };
+            if (strict) {
+              throw new StopDecodingError(error);
+            }
```

Comment:
Does this throw need to be guarded by strict? The preceding comment block and this code is confusing. It's also likely I need to re-read your PR explanation :/

Reply:
Yes, all the throws need to be guarded by strict, because that's how the decoder works; when something goes wrong, its response depends on whether it's in strict mode or loose mode.  In strict mode it throws an error, but in loose mode, it just returns an `ErrorResult` instead.

Strict mode is used for things that could possibly have multiple interpretations; it's used for decoding events and for decoding returned data.  Loose mode is used for things that should only have one interpretation; it's used for decoding transactions and state variables.

Reply:
Thanks!